### PR TITLE
Account page and management improvements

### PR DESCRIPTION
**Pull request by CTalvio**
## Pull Request Description

- Remove logout button from account page
- Move the manage accounts button to the left (where log out used to be)
- Enable logging out from the last account from the manage accounts modal
- Do not display "you are browsing anonymously" placeholder when account is actually loading
- Do not display account page before it has fully loaded (before, an empty view was momentarily displayed)
- Change instance links in login dialogues to outline buttons
- Add "log out" to general account actions on account setting page
- Add "manage accounts" dialogue to account setting page (this also replaces the "you are browsing anonymously" placeholder when an anonymous account is used or is the only account available)

## Issue Being Fixed

Various inconsistencies on the account page, alleviates confusion about where to find what and what the state of the account page is. Makes common actions available where one would think to look, especially when looking for account management via the settings tab. (before, you would never have found a way to log out from here, or add/remove accounts)

## Screenshots / Recordings

image

image

Here's what switching accounts looked like before:
Screen_recording_20240506_224508.webm

After:
Screen_recording_20240506_224542.webm

## Checklist

- [ ] Did you update CHANGELOG.md?
- [x] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

File: lib/user/pages/user_settings_page.dart [hunk 1]
```diff
@@ -129,169 +126,197 @@ class _UserSettingsPageState extends State<UserSettingsPage> {
...
+                          onTap: () => showProfileModalSheet(context)),
+                      if (state.status != UserSettingsStatus.notLoggedIn)
+                        Column(
+                          crossAxisAlignment: CrossAxisAlignment.stretch,
+                          mainAxisAlignment: MainAxisAlignment.start,
+                          children: [
+                            const Divider(),
+                            Padding(
+                              padding: const EdgeInsets.symmetric(horizontal: 16.0, vertical: 8.0),
+                              child: Text('This Account', style: theme.textTheme.titleMedium),
```
Comment (micahmo) [inline lib/user/pages/user_settings_page.dart:145]: Don't forget to localize this!

**Review by micahmo — COMMENTED** [review]
Just wanted to say, thank you so much for doing this! A lot of these things have been bugging me as I'm sure they've been bugging you (like the browsing anonymously thing when switching accounts).

From a UI/UX perspective, these are my only comments.
* When using the in-app browser and navigating to one of the buttons (e.g., "Open Instance") the in-app browser is shown in the modal and is not scrollable. Most likely you didn't introduce this problem, so we can fix it later.
* This is super nit-picky, but do you want to match the horizontal rule in the account settings page with the ones on the Debug and Appearance page (or vice versa) just for consistency?
* If the current account is going to be switchable via "Manage Accounts" on the account settings page, should we make it *not* switchable by tapping the current account (if it feels redundant to you)? If so, that could revert from a `UserSelector` to a `UserIndicator`.
* If I remove all accounts and then navigate to Settings > Account, I get an infinite spinner. Previously it would display the "you are browsing anonymously" screen, from which you could log in without leaving that page.

https://github.com/thunder-app/thunder/assets/7417301/f840cc5f-ea32-41ac-84c5-c3345d80528c

Otherwise everything else LGTM!

**Comment by CTalvio** [comment]
> If I remove all accounts and then navigate to Settings > Account, I get an infinite spinner.

Good catch. And agreed on using a user indicator, the switchers are a bit redundant but I didn't bother figuring out something better, and what a timesave that was, as you now reveal the perfect widget exists already :D

> This is super nit-picky, but do you want to match the horizontal rule in the account settings page with the ones on the Debug and Appearance page (or vice versa) just for consistency?

Yes. This would absolutely have bothered me the second I noticed.

**Comment by micahmo** [comment]
Hi @CTalvio, sorry for being annoying! Your latest commit does fix the account settings page from settings! But now there seems to be an issue with the profile page.

1. If I log out of the last account, it leaves my account info there.
2. If I refresh, it gets stuck spinning forever.

https://github.com/thunder-app/thunder/assets/7417301/51ceceba-a71c-4b88-85e5-3b0b28f453b5

**Comment by CTalvio** [comment]
That seems to do the trick.

I'm not sure what the returns in those listeners were there to do?

**Comment by micahmo** [comment]
Oh, we can't take those out! Those returns are what supports the temporary account feature (where you can post or comment as a different user). It allows us to switch the account without reloading the whole app. When `reload` is `false`, we shouldn't rebuild parts of the app like the user page or the feed. Sorry if that makes your feature more complicated (although I'm not sure why `reload` is `false` in your case).

**Comment by CTalvio** [comment]
Yeah I figured they were there for a reason... Hence my commenting on it.

The account page is leaving the old account page up because it doesn't update on changes elsewhere.

**Comment by CTalvio** [comment]
I'll poke at it some more, but if nothing else I'll obviously reverse removing those returns.

**Comment by micahmo** [comment]
> This is super nit-picky, but do you want to match the horizontal rule in the account settings page with the ones on the Debug and Appearance page (or vice versa) just for consistency?

Just a note, I'm refactoring this divider widget (since we're using it a bunch now) in #1359, so whichever one of us goes first can pick up the changes from the other. 😊

**Comment by hjiangsu** [comment]
I was playing around with this a bit, and noticed that if you manage accounts from the account settings, it changes your entire profile (until you refresh again?). Previously, switching users in the account settings would only temporarily switch users.

Here's what I mean:

https://github.com/thunder-app/thunder/assets/30667958/a488eec1-7f74-409c-ad77-369efe1f13b7

https://github.com/thunder-app/thunder/assets/30667958/81123bb0-5704-4c7d-b4c6-24e0db57eda0

**Comment by CTalvio** [comment]
> It's behaving weird in general. Switching the account while in account settings doesn't stick, but doing so from the account page, does. Removing the returns doesn't fix that.

It's supposed to actually change the account, unlike working the same way as posting/commenting as a different user like it did before. But it doesn't stick for some reason, as you point out.

**Comment by CTalvio** [comment]
Mystery solved, I didn't remove the code that restores previous account upon popping out of the account settings.

As for the account settings sticking around when going from logged-in to accounts removed/anonymous due to the reload not happening for some reason, adding a logged-in check to the reload returns works around that oddity.

**Review by hjiangsu — APPROVED** [review]
LGTM - just one nitpick

File: lib/user/pages/user_settings_page.dart [hunk 2]
```diff
@@ -24,11 +23,13 @@ import 'package:thunder/shared/avatars/user_avatar.dart';
 import 'package:thunder/thunder/thunder_icons.dart';
 import 'package:thunder/user/bloc/user_settings_bloc.dart';
 import 'package:thunder/user/utils/restore_user.dart';
-import 'package:thunder/user/widgets/user_selector.dart';
+import 'package:thunder/user/widgets/user_indicator.dart';
 import 'package:thunder/utils/instance.dart';
 import 'package:thunder/utils/links.dart';
 import 'package:thunder/instance/utils/navigate_instance.dart';
 
+import '../../account/utils/profiles.dart';
```
[the remark this answers is not in the document]
Comment (CTalvio) [reply lib/user/pages/user_settings_page.dart]: ack! I need to get into the habit of checking that, I wonder if I can set AS to respect that when it auto-adds imports.

**Pull request merged**